File: 041-和为S的连续正数序列/041-FindContinuousSequence.py
# -*- coding:utf-8 -*-
class Solution:
    def FindContinuousSequence(self, tsum):
        # write code here
        if tsum<3:
            return []
        small=1
        big=2
        csum=small+big
        ans=[]
        while small<(1+tsum)//2:
            if csum==tsum:
                ans.append(list(range(small,big+1)))
                big+=1
                csum+=big
            elif csum<tsum:
                big+=1
                csum+=big
            else:
                csum-=small
                small+=1
        return ans

File: 041-和为S的连续正数序列/041-FindContinuousSequence.cpp
class Solution {
public:
    vector<vector<int> > FindContinuousSequence(int sum) {
        vector<vector<int>> ans;
        if(sum<3){
            return ans;
        }
        int small=1;
        int big=2;
        int mid=(1+sum)/2;
        int csum=small+big;
        while (small<mid){
            if(csum==sum){
                vector<int> temp;
                for(int i=small;i<=big;i++)
                {
                    temp.push_back(i);
                }
                ans.push_back(temp);
                big++;
                csum+=big;
            }
            else if(csum<sum){
                big++;
                csum+=big;
            }
            else{
                csum-=small;
                small++;
            }
        }
        return ans;
    }
};
